src/update.c

```diff
@@ -555,6 +555,15 @@ int main_update()
 	run_scripts();
 	grabtime_stop(&times);
 
+	/* Create the state file that will tell swupd it's on a mix on future runs */
+	if (mix_exists && !system_on_mix()) {
+		int fd = open("/usr/share/defaults/swupd/mixed", O_RDWR | O_CREAT, S_IRWXU | S_IRGRP | S_IROTH);
```

Comment:
Probably need to add /usr/share/defaults/swupd to create_required_dirs().  And/or enforce in swupd-server and mixer that it exists.